t/29overload.t

```diff
@@ -110,6 +110,16 @@ use DateTime;
         'Cannot compare a DateTime object to a FooBar object'
     );
 
+    my $saw_undef_cmp_warning = 0;
+    $SIG{__WARN__} =
+        sub { $saw_undef_cmp_warning = 1 if $_[0] =~ /uninitialized value in numeric/ };
+    { my $x = undef > $dt; }
+    $SIG{__WARN__} = undef;
+    ok(
+        $saw_undef_cmp_warning,
+        'Comparing undef to a DateTime object generates a Perl warning'
+    );
+
```

Comment:
I'd prefer that you use [Test::Warnings](https://metacpan.org/pod/Test::Warnings) here.